src/courses/views.py

```diff
@@ -167,10 +167,32 @@ def get_form_kwargs(self):
         kwargs['instance'] = CourseStudent(user=user)
         return kwargs
 
+    def get_context_data(self, **kwargs):
+        ctx = super().get_context_data()
+        ctx['heading'] = 'Add student to course'
+        ctx['submit_btn_value'] = 'Add'
+        return ctx
+
     def get_success_url(self):
         return reverse('profiles:profile_detail', args=(self.object.user.profile.id, ))
 
 
+@method_decorator(staff_member_required, name='dispatch')
+class CourseStudentUpdate(NonPublicOnlyViewMixin, UpdateView):
+    model = CourseStudent
+    form_class = CourseStudentForm
```

Comment:
Might need to extend CourseStudentForm with a few extra fields for the staff edit form.